src/main/resources/public/peer.js

```diff
@@ -0,0 +1,257 @@
+  connectToSessionButton.onclick = event => {
+    if (connectToSessionForm.checkValidity()) {
+      // Allow HTML5 form validation to take place
+      event.preventDefault()
+      startDwrtc(true, initialOtherPeerIdInput.value)
+      hideIdMessage()
+    }
+  }
+  idCopy.onclick = event => {
+    event.preventDefault()
+    idValue.select()
+    document.execCommand("copy")
+    idCopy.textContent = "Copied!"
+  }
+  enableInput()
+}
+
+const showOutput = () => {
+  output.hidden = false
+  // if the css is "display: grid" initially, this overrides the hidden attribute
+  // therefore, we have to unhide it and add the proper class
+  output.classList.add("grid")
+}
+
+const hideIdMessage = () => {
+  idMessage.hidden = true
+}
+
+const showOtherVideo = () => {
+  otherVideo.hidden = false
+  hideIdMessage()
+}
+
+const copyIdToClipboard = () => {
+  idValue.select()
+  document.execCommand("copy")
+}
+
+const enableInput = () => (input.hidden = false)
+
+const disableInput = () => (input.hidden = true)
+
+const showError = error => {
+  errorOverlay.hidden = false
+  errorOverlay.classList.add("fade-in")
+  errorMessage.textContent = error
+}
+
+/**
+ * Called when the user is ready. Sets up DWRTC.
+ */
+async function startDwrtc(initiator, initialPeerId) {
+  console.log("Start DWRTC")
+  disableInput()
+  showOutput()
+  const dwrtc = new DWRTC(initiator, initialPeerId)
+  await dwrtc.setup()
+}
+
+class DWRTC {
+  constructor(isInitiator, initialPeerId) {
+    this.isInitiator = isInitiator
+    if (this.isInitiator) {
+      this.otherPeerId = initialPeerId
+    }
+    console.log(
+      `Started DWRTC with isInitiator: ${isInitiator}, initialPeerId: ${initialPeerId}`
+    )
+  }
+
+  /**
+   * Sets the class up
+   * This is a separate method to the constructor, since constructors are not allowed to be async
+   */
+  async setup() {
+    await this.startSimplePeer()
+    await this.setupSocket()
+    this.completeSimplePeerSetup()
+  }
+
+  /**
+   * Initialize the peer with the information we have so far
+   */
+  async startSimplePeer() {
+    try {
+      const stream = await navigator.mediaDevices.getUserMedia({
+        video: true,
+        audio: true
+      })
+      yourVideo.srcObject = stream
+      yourVideo.play()
+      yourVideo.muted = true
+      this.peer = new window.SimplePeer({
+        initiator: this.isInitiator,
+        stream: stream
+      })
+    } catch (error) {
+      throw error
+    }
+    // TODO https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia
+    // "It's possible for the returned promise to neither resolve nor reject,
+    // as the user is not required to make a choice at all and may simply ignore the request."
+    console.debug("SimplePeer started")
+  }
+
+  /** Initialize the websocket completely */
+  async setupSocket() {
+    this.socket = new WebSocket(WEBSOCKET_URL)
+
+    // Dummy promise that we can resolve when the websocket is open
+    await new Promise(
```

Comment:
This it to wait until the socket is open? If yes I'd suggest to clarify a bit in the comment:
Dummy promise to wait until the socket is opened.

I'd probably move it to a seperate function to make it clearer. You also wouldn't need a comment then.
https://stackoverflow.com/a/21394730/1765500